docs/src/userguide/language_basics.rst

```diff
@@ -1059,8 +1060,13 @@ Some things to note:
   function returning nothing (C ``void``).  Simple workarounds are to mark the
   function as ``noexcept`` if you're certain that exceptions cannot be thrown, or
   to change the return type to ``int`` and just let Cython use the return value
-  as an error flag (by default, ``-1`` triggers the exception check).
+  as an error flag (by default, ``-1`` triggers the exception check, i.e.
+  functions returning ``int`` are ``except? -1`` by default)
 
+  When the return value is a C type, and the last statement is executed
+  without returning a value, the function returns ``0`` (instead of ``None``).
+  However, explicit ``return`` in the function body is not allowed, you need
+  explicit ``return 0``.
```

Comment:
Hmm. This section is about exception propagation. I don't think this is the best place to put this paragraph.